bazel/candid.bzl

```diff
@@ -14,16 +14,18 @@ if [[ $mr_title == *"[override-didc-check]"* ]]; then
     exit 0
 fi
 
-# https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
-# The HEAD SHA of the target branch of the merge request. The variable is empty in merge request pipelines. 
-# The SHA is present only in merged results pipelines.
-readonly merge_base=${{CI_MERGE_REQUEST_TARGET_BRANCH_SHA:-HEAD}}
+if [ -z "${{CI_MERGE_REQUEST_TARGET_BRANCH_NAME:-}}" ]; then
+  readonly target_branch_name="HEAD"
+else
+  readonly target_branch_name="origin/$CI_MERGE_REQUEST_TARGET_BRANCH_NAME"
+fi
+readonly merge_base="$(git merge-base "$target_branch_name" HEAD)"
 
 readonly tmpfile=$(mktemp $TEST_TMPDIR/prev.XXXXXX)
 readonly errlog=$(mktemp $TEST_TMPDIR/err.XXXXXX)
 
 if ! git show $merge_base:{did_path} > $tmpfile 2> $errlog; then
-    if grep -sq -- "exists on disk, but not in \\|does not exist in 'HEAD'" $errlog; then
+    if grep -sq -- "does not exist in" $errlog; then
```

Comment:
If you try my above suggestion, can you try keeping this line as it was and see if it passes? It should check for one of either messages.

Reply:
Trying ...

Reply:
Works.